ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -146,7 +188,10 @@ public Topology getTopology() {
 
   public Collection<StreamsMetadata> getAllMetadata() {
     try {
-      return kafkaStreams.allMetadata();
+      // Synchronized block need until https://issues.apache.org/jira/browse/KAFKA-9668 fixed.
+      synchronized (streamsMetadataState) {
+        return kafkaStreams.allMetadata();
```

Comment:
Isn't it good practice to return an immutable list here?

Reply:
Yep, good catch.